src/Illuminate/Validation/Concerns/ValidatesAttributes.php

```diff
@@ -819,6 +819,7 @@ public function parseTable($table)
         [$connection, $table] = Str::contains($table, '.') ? explode('.', $table, 2) : [null, $table];
 
         if (Str::contains($table, '\\') && class_exists($table) && is_a($table, Model::class, true)) {
+            $connection = $connection ?? (new $table)->getConnectionName();
             $table = (new $table)->getTable();
         }
```

Comment:
Can't we do this to prevent creating multiple instances of the model?

```
$model = new $table;

$table = $model->getTable();
$connection = $connection ?: $model->getConnection();
```

Reply:
@themsaid good point - have now made this change and committed here.